handling-forms/submitting-forms.md

```diff
@@ -0,0 +1,428 @@
+
+<!-- prettier-ignore-start -->
+```js
+import PropTypes from 'prop-types';
+import './Student.css';
+
+const Student = (props) => {
+
+    const onAttendanceButtonClick = () => {
+        const updatedStudent = {
+            id: props.id,
+            nameData: props.name,
+            emailData: props.email,
+            isPresentData: !props.isPresent
+        }
+
+        // Invoke the function passed in through the prop named "onUpdate"
+        // This function is referenced by the name "updateStudentData" in App
+        props.onUpdate(updatedStudent);
+    }
+
+    const nameColor = props.isPresent ? 'green' : 'red';
+
+    return (
+        <div>
+            <ul>
+                <li className={nameColor}>Nickname: {props.name}</li>
+                <li>Email: {props.email}</li>
+            </ul>
+            <button onClick={onAttendanceButtonClick}>Toggle if {props.name} is present</button>
+        </div>
+    )
+}
+
+Student.propTypes = {
+    id: PropTypes.number.isRequired,
+    name: PropTypes.string.isRequired,
+    email: PropTypes.string.isRequired,
+    isPresent: PropTypes.bool,
+    onUpdate: PropTypes.func.isRequired
+};
+
+export default Student;
+```
+<!-- prettier-ignore-end -->
+
+</details>
+
+<details>
+
+<summary>A <code>NewStudentForm</code> component, responsible for containing a form to add a new student.</summary>
+
+<!-- prettier-ignore-start -->
+```js
+import { useState } from 'react';
+
+const NewStudentForm = () => {
+
+    const [formFields, setFormFields] = useState({
+        name: '',
+        email: ''
+    });
+
+    const onNameChange = (event) => {
+        setFormFields({
+            ...formFields,
+            name: event.target.value
+        })
+    };
+
+    const onEmailChange = (event) => {
+        setFormFields({
+            ...formFields,
+            email: event.target.value
+        })
+    };
+
+    return (
+        <form>
+            <div>
+                <label htmlFor="fullName">Name:</label>
+                <input
+                    name="fullName"
+                    value={formFields.name}
+                    onChange={onNameChange}/>
+            </div>
+            <div>
+                <label htmlFor="email">Email:</label>
+                <input name="email"
+                    value={formFields.email}
+                    onChange={onEmailChange}/>
+            </div>
+            <input
+                type="submit"
+                value="Add Student"
+            />
+        </form>
+    );
+}
+
+export default NewStudentForm;
+```
+<!-- prettier-ignore-end -->
+
+</details>
+
+The `NewStudentForm` component is all well and good. It follows the controlled component pattern, so its state is in sync with the input fields.
+
+However, all student data is managed in the `App` component. Our new student form doesn't actually add a student!
+
```

Comment:
```suggestion

<!-- TODO: It would be great to have a screenshot here of the aftermath of trying to click the add button, only to have the form submission occur. -->

```

Reply:
Created an issue for this